File: fixtures/consumers/logistics-service/src/order_flow.py
from shipments_client import submit_shipment, is_shipment_queued, is_shipment_delivered


def place_order(order):
    request = {
        "originZip": order["originZip"],
        "destZip": order["destZip"],
        "weightKg": order["weightKg"],
        "carrier": "ups",
    }
    shipment = submit_shipment(request)
    if is_shipment_queued(shipment):
        return shipment
    if is_shipment_delivered(shipment):
        return shipment
    return shipment

File: fixtures/consumers/logistics-service/src/order_flow.go
package logistics

func PlaceOrder(order map[string]interface{}) map[string]interface{} {
	request := map[string]interface{}{
		"originZip": order["originZip"],
		"destZip":   order["destZip"],
		"weightKg":  order["weightKg"],
		"carrier":   "ups",
	}
	shipment := SubmitShipment(request)
	if IsShipmentQueued(shipment) {
		return shipment
	}
	if IsShipmentDelivered(shipment) {
		return shipment
	}
	return shipment
}
